### PR TITLE
Add a service with local internal traffic policy for traces and dogstatsd.

### PR DESCRIPTION
**Pull request by L3n41c**
#### What this PR does / why we need it:

Add a service which leverages [internal traffic policy](https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/) for traces and dogstatsd.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [X] Chart Version bumped
- [X] `CHANGELOG.md` has beed updated
- [X] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/templates/_helpers.tpl [hunk 1]
```diff
@@ -503,3 +503,14 @@ true
...
+{{/*
+Return true if we can enable Service Internal Traffic Policy
+*/}}
+{{- define "enable-service-internal-traffic-policy" -}}
+{{- if semverCompare "^1.22-0" .Capabilities.KubeVersion.GitVersion -}}
+true
+{{- else -}}
+false
+{{- end -}}
+{{- end -}}
```
Comment (clamoriniere) [inline charts/datadog/templates/_helpers.tpl:516]:
```suggestion
{{- end -}}

```

**Review by clamoriniere — COMMENTED** [review]
just need to rebase and bump the version again

**Comment by vboulineau** [comment]
It should replace the existing `serviceTopology` bits: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L289-L303

**Comment by L3n41c** [comment]
> It should replace the existing serviceTopology bits: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L289-L303

I hesitated. Is it OK to completely get rid of that ? And bump the major version as this removal wouldn’t be backward compatible ?

**Comment by vboulineau** [comment]
> > It should replace the existing serviceTopology bits: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L289-L303
> 
> I hesitated. Is it OK to completely get rid of that ? And bump the major version as this removal wouldn’t be backward compatible ?

As it's deprecated and never made it to `beta` I'd say yes.

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**